src/Dapr.Client/DaprClient.cs

```diff
@@ -939,6 +940,286 @@ public async Task<StateEntry<TValue>> GetStateEntryAsync<TValue>(string storeNam
+        /// <param name="encryptionOptions">Options informing how the encryption operation should be configured.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of encrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract Task<ReadOnlyMemory<byte>> EncryptAsync(string vaultResourceName,
+            ReadOnlyMemory<byte> plaintextBytes, string keyName, EncryptionOptions encryptionOptions,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Encrypts a stream using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="plaintextStream">The stream containing the bytes of the plaintext value to encrypt.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the encryption operation.</param>
+        /// <param name="encryptionOptions">Options informing how the encryption operation should be configured.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of encrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract Task<ReadOnlyMemory<byte>> EncryptAsync(string vaultResourceName, Stream plaintextStream,
+            string keyName, EncryptionOptions encryptionOptions,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Encrypts a stream using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="plaintextStream">The stream containing the bytes of the plaintext value to encrypt.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the encryption operation.</param>
+        /// <param name="encryptionOptions">Options informing how the encryption operation should be configured.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of encrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract IAsyncEnumerable<byte[]> EncryptStreamAsync(string vaultResourceName, Stream plaintextStream, string keyName,
```

Comment:
`IAsyncEnumerable<ReadOnlyMemory<byte>>`?

Reply:
A `Stream` works from an API perspective but, AFAIK, there's no built-in `Stream` implementation that provides the needed semantics (having a reader read from the stream while another task writes chunks to the stream).  (There are other .NET mechanisms, such as blocks/channels, that provide those semantics but they typically don't find their ways into common or third-party APIs.)  Returning a `Stream` would mean implementing a new type and I don't know whether it's worth the effort.

The difference in usage would be something like:

```charp

await using var inputStream = ...;
await using var outputStream = ...;

// Stream-returning API

await using var encryptedDataStream = await daprClient.EncryptStreamAsync(..., inputStream, ...);

await encryptedDataStream.CopyToAsync(outputStream);

// Async-enumerable-returning API

var encryptedData = await daprClient.EncryptStreamAsync(..., inputStream, ...);

await foreach (var chunk in encryptedData)
{
  await outputStream.WriteAsync(chunk);
}

```

Reply:
Yeah, we didn't have a suitable built-in Stream implementation in JS either.

The possible advantage of using something that implements the Stream API is that it allows composition. For example in JS (and Go) there are built-in methods to pipe streams (for example, open a file in write mode, and then pipe from the output stream into the file directly), etc. It allows for more idiomatic APIs.

I also don't know if this is the case in C#, but in Go and JS using the stream APIs allows managing backpressure too, all the way to the gRPC channel.

Reply:
There's little reason the developer can't enjoy the relative magic of `await foreach` and drop the result into a stream (or buffer or yield return out to something else) as they'd like. I just updated the examples to reflect encrypting data from a `FileStream` to an in-memory buffer via a `List<byte>` and then decrypting via a `MemoryStream` from that list back into another `FileStream`.

Reply:
Ok. Make sure that handles backpressure too please!

Reply:
Updated methods to use `IAsyncEnumerable<ReadOnlyMemory<byte>>` along with a refresh to the examples.